examples/json-web-token.roc

```diff
@@ -0,0 +1,105 @@
+app [Model, server] { pf: platform "../platform/main.roc" }
+
+import pf.Stdout
+import pf.Http exposing [Request, Response]
+import pf.Jwt
+
+Model : {}
+
+server = { init, respond }
+
+respond : Request, Model -> Task Response [ServerErr Str]_
+respond = \_, _ ->
+
+    # We hardcode the JWT here and ignore the request for simplicity, but normally this is how
+    # you would decode the request body (assuming the token is in the body and not a URL param)
+    # ```
+    # exampleBody = Http.parseFormUrlEncoded req.body
+    # ```
+    exampleBody : Result (Dict Str Str) _
+    exampleBody =
+        Dict.empty {}
+        |> Dict.insert "id_token" "eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJzdWIiOiIxMjM0NTY3ODkwIiwibmFtZSI6IkpvaG4gRG9lIiwiaWF0IjoxNTE2MjM5MDIyfQ.joReqPNNkWQ8zQCW3UQnhc_5NMrSZEOQYpk6sDS6Y-o"
+        |> Ok
+
+    token : Str
+    token =
+        exampleBody
+        |> Result.try \decodedBody -> decodedBody |> Dict.get "id_token"
+        |> Result.withDefault "TOKEN MISSING"
+
+    when Jwt.verify { secret: "shhh_very_secret", algorithm: Hs256, token } |> Task.result! is
+        Err (JwtErr err) ->
+            Task.ok {
+                status: 200,
+                headers: [
+                    { name: "Content-Type", value: "text/plain" },
+                ],
+                body: Str.toUtf8 "UNABLE TO DECODE JWT\n$(Inspect.toStr err)\n",
+            }
+
+        Ok claims ->
+            sub = claims |> Dict.get "sub" |> Result.withDefault "SUBJECT CLAIM MISSING"
```

Comment:
Can we cleanup all of these `Result.withDefault` calls? I think this would be a great place to show off how to merge result errors into tasks and handle them properly.

Reply:
Done in e8c55628c3042953f02061ead6d00229547d2cc1